sdk/python/feast/infra/redis_provider.py

```diff
@@ -0,0 +1,276 @@
+        self,
+        feature_view: FeatureView,
+        start_date: datetime,
+        end_date: datetime,
+        registry: Registry,
+        project: str,
+        tqdm_builder: Callable[[int], tqdm],
+    ) -> None:
+        entities = []
+        for entity_name in feature_view.entities:
+            entities.append(registry.get_entity(entity_name, project))
+
+        (
+            join_key_columns,
+            feature_name_columns,
+            event_timestamp_column,
+            created_timestamp_column,
+        ) = _get_column_names(feature_view, entities)
+
+        start_date = utils.make_tzaware(start_date)
+        end_date = utils.make_tzaware(end_date)
+
+        offline_store = get_offline_store_from_sources([feature_view.input])
+        table = offline_store.pull_latest_from_table_or_query(
+            data_source=feature_view.input,
+            join_key_columns=join_key_columns,
+            feature_name_columns=feature_name_columns,
+            event_timestamp_column=event_timestamp_column,
+            created_timestamp_column=created_timestamp_column,
+            start_date=start_date,
+            end_date=end_date,
+        )
+
+        if feature_view.input.field_mapping is not None:
+            table = _run_field_mapping(table, feature_view.input.field_mapping)
+
+        join_keys = [entity.join_key for entity in entities]
+        rows_to_write = _convert_arrow_to_proto(table, feature_view, join_keys)
+
+        with tqdm_builder(len(rows_to_write)) as pbar:
+            self.online_write_batch(
+                project, feature_view, rows_to_write, lambda x: pbar.update(x)
+            )
+
+        feature_view.materialization_intervals.append((start_date, end_date))
+        registry.apply_feature_view(feature_view, project)
+
+    def _get_cs(self):
+        """
+        Reads Redis connections string using format
+        for RedisCluster:
+            redis1:6379,redis2:6379,decode_responses=true,skip_full_coverage_check=true,ssl=true,password=...
+        for Redis:
+            redis_master:6379,db=0,ssl=true,password=...
+        """
+        connection_string = os.environ["REDIS_CONNECTION_STRING"]
+        startup_nodes = [
+            dict(zip(["host", "port"], c.split(":")))
+            for c in connection_string.split(",")
+            if "=" not in c
+        ]
+        params = {}
+        for c in connection_string.split(","):
+            if "=" in c:
+                kv = c.split("=")
+                try:
+                    kv[1] = json.loads(kv[1])
+                except json.JSONDecodeError:
+                    ...
+
+                it = iter(kv)
+                params.update(dict(zip(it, it)))
+
+        return startup_nodes, params
+
+    def _get_client(self):
+        """
+        Creates the Redis client RedisCluster or Redis depending on configuration
+        """
+        startup_nodes, kwargs = self._get_cs()
+
+        if os.environ["REDIS_TYPE"] == "REDIS_CLUSTER":
```

Comment:
We certainly should be moving to using purely `feature_store.yaml` for configuration, and not environmental variables. It will be too hard to keep track of these configuration backdoors.

Reply:
Does `feature_store.yaml` support provider-specific configuration? If it does, I agree that using it may be better than environment variables, except the connection string, of course. I assume the project YAML is supposed to be stored with the project in git, etc., so you wouldn't add connection details there?

Reply:
Yes, that is the goal. Also, I think we can add environmental variable loading when parsing the `feature_store.yaml` file, or optionally make it possible to set values inside the file using env vars. That way we are keeping track of all options in a single place, and loading of env vars happens during initialization, not in a place deeply nested in the code.

Reply:
Ah, that's a good idea! I agree that it's best to have a single place that all of the configuration options are visible.